src/embind/embind_ts.js

```diff
@@ -337,6 +352,30 @@ var LibraryEmbind = {
       classDef.methods.push(funcDef);
     });
   },
+  _embind_register_class_property__deps: [
+    '$readLatin1String', '$whenDependentTypesAreResolved', '$ClassProperty'],
+  _embind_register_class_property: function(classType,
+                                            fieldName,
+                                            getterReturnType,
+                                            getterSignature,
+                                            getter,
+                                            getterContext,
+                                            setterArgumentType,
+                                            setterSignature,
+                                            setter,
+                                            setterContext) {
+    fieldName = readLatin1String(fieldName);
+    assert(getterReturnType === setterArgumentType, 'Mismatched getter and setter types are not supported.');
+    whenDependentTypesAreResolved([], [classType], function(classType) {
+      classType = classType[0];
+      whenDependentTypesAreResolved([], [getterReturnType], function(types) {
```

Comment:
Can these be arrow functions?

Reply:
Yeah, this file could use arrow conversion overall. I'll do that in a separate PR.

Reply:
SGTM